src/components/input/PasswordInput.tsx

```diff
@@ -0,0 +1,42 @@
+'use client';
+
+import * as React from 'react';
+import { cn } from '@/lib/utils';
+import { Input, InputProps } from '../ui/input';
+import { Eye, EyeOff } from 'lucide-react';
+
+export interface PasswordInputProps extends InputProps {}
+
+const PasswordInput = React.forwardRef<HTMLInputElement, PasswordInputProps>(
+  ({ className, ...props }, ref) => {
+    const [showPassword, setShowPassword] = React.useState(false);
```

Comment:
```suggestion
  const [showPassword, togglePasswordVisibility] = useReducer((state) => !state, false);
```
위처럼 선언하면 togglePasswordVisibility를 따로 함수로 만들어주지 않아도 됩니다!
코드가 간결해져요!!

https://www.youtube.com/shorts/XBNOEHA4Kvg